bin/scripts/continious-data-population.js

```diff
@@ -0,0 +1,1861 @@
+        for (var i = 0; i < Object.keys(inputs).length; i++) {
+            pluginList[inputs[i].APP_ID] = [];
+            app[inputs[i].APP_ID] = await db.collection('apps').findOne({'_id': db.ObjectID(inputs[i].APP_ID)});
+            if (inputs[i].POPULATOR_TEMPLATE_ID.length > 18) {
+                template[inputs[i].APP_ID] = await db.collection('populator_templates').findOne({'_id': db.ObjectID(inputs[i].POPULATOR_TEMPLATE_ID)});
+            }
+            else { // default template
+                defaultTemplates.forEach(function(item) {
+                    if (item._id === inputs[i].POPULATOR_TEMPLATE_ID) {
+                        template[inputs[i].APP_ID] = item;
+                    }
+                });
+                if (!template[inputs[i].APP_ID]) {
+                    db.close();
+                    writeMsg('INVALID POPULATOR_TEMPLATE_ID ', inputs[i].POPULATOR_TEMPLATE_ID);
+                }
+            }
+            if (!app[inputs[i].APP_ID]) {
+                db.close();
+                writeMsg('INVALID APP_ID ', inputs[i].APP_ID);
+            }
+            if (!template) {
+                db.close();
+                writeMsg('INVALID POPULATOR_TEMPLATE_ID ', inputs[i].POPULATOR_TEMPLATE_ID);
+            }
+            db.close();
+            users[inputs[i].APP_ID] = [];
+
+
+            pluginList[inputs[i].APP_ID] = await sendRequest({
+                requestType: 'GET',
+                Url: SERVER_URL + "/o/plugins?",
+                requestParamKeys: { app_id: inputs[i].APP_ID, api_key: API_KEY }
+            });
+        }
+    }
+    catch (error) {
+        clearInterval(interval);
+        writeMsg('There was an error while checking db parameters', error);
+    }
+}
+readDbParameters();
+
+var interval = setInterval(async function() {
+    for (var z = 0; z < Object.keys(inputs).length; z++) {
+        //users = [];
+        var userCount = getRandomInt(maxUserCount / 2, maxUserCount);
+        if (!users[inputs[z].APP_ID] || users[inputs[z].APP_ID].length === 0) {
+            for (var i = 0; i < userCount; i++) {
+                user = new getUser(template[inputs[z].APP_ID] && template[inputs[z].APP_ID].up);
+                users[inputs[z].APP_ID].push(user);
+            }
+        }
+        else {
+            if (users[inputs[z].APP_ID].length < maxUserCount) {
+                if (getRandomInt(1, 3) === 1) {
+                    user = new getUser(template && template.up);
+                    users[inputs[z].APP_ID].push(user);
+                }
+            }
+        }
+
+        var upperBoundary = Math.floor(userCount / 10);
+
+        var indexes = [];
+        for (var i0 = 0; i0 < upperBoundary; i0++) {
+            indexes.push(getRandomInt(0, users[inputs[z].APP_ID].length - 1));
+        }
+
+        indexes.forEach(function(index) {
+            if (users[inputs[z].APP_ID][index].hasSession) {
+                if (users[inputs[z].APP_ID][index].eventsInSession >= 1) {
```

Comment:
Lets keep this like you had:
(users[inputs[z].APP_ID][index].eventsInSession >=9)
So it would mean  at least 9 events in session